internal/service/advancedclustertpf/resource.go

```diff
@@ -99,80 +96,230 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
-		err := StoreUpdatePayloadProcessArgsLegacy(patchReqProcessArgsLegacy)
+		legacyAdvConfig, _, err = r.Client.AtlasV220240530.ClustersApi.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, patchReqProcessArgsLegacy).Execute()
 		if err != nil {
-			diags.AddError("error storing update payload advanced config legacy", fmt.Sprintf("error storing update payload: %s", err.Error()))
+			diags.AddError("errorUpdateAdvancedConfigLegacy", fmt.Sprintf(errorConfigUpdate, clusterName, err.Error()))
 		}
 	}
-	if diags.HasError() {
-		return
-	}
-
-	// TODO: Use requests to do actual updates with Admin API
-	tfNewModel, shouldReturn := mockedSDK(ctx, diags, plan.Timeouts)
-	// TODO: keep project_id and name from plan to avoid overwriting for move_state tests. We should probably do the same with the rest of attributes
-	tfNewModel.Name = plan.Name
-	tfNewModel.ProjectID = plan.ProjectID
-	if shouldReturn {
-		return
+	if cluster != nil {
+		r.convertClusterAddAdvConfig(ctx, legacyAdvConfig, advConfig, cluster, plan.Timeouts, diags, &resp.State)
+	} else {
+		r.readCluster(ctx, &plan, &resp.State, diags)
 	}
-	diags.Append(resp.State.Set(ctx, tfNewModel)...)
 }
-
 func (r *rs) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
 	var state TFModel
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-	if resp.Diagnostics.HasError() {
+	diags := &resp.Diagnostics
+	diags.Append(req.State.Get(ctx, &state)...)
+	if diags.HasError() {
 		return
 	}
-	name := state.Name.ValueString()
-	clusterID := state.ClusterID.ValueString()
-	if clusterID == "" {
-		resp.Diagnostics.AddError("errorDelete", fmt.Sprintf(errorDelete, name, "clusterID is empty"))
+	clusterName := state.Name.ValueString()
+	projectID := state.ProjectID.ValueString()
+	if clusterName == "" {
+		resp.Diagnostics.AddError("errorDelete", fmt.Sprintf(errorDelete, clusterName, "clusterName is empty"))
 		return
 	}
+	api := r.Client.AtlasV2.ClustersApi
+	_, err := api.DeleteCluster(ctx, projectID, clusterName).Execute()
+	if err != nil {
+		diags.AddError("errorDelete", fmt.Sprintf(errorDelete, clusterName, err.Error()))
+	}
+	_, summary, detail := r.awaitChanges(ctx, &state.Timeouts, diags, projectID, clusterName, "delete")
+	if summary != "" || detail != "" {
+		diags.AddError(summary, detail)
+	}
 }
 
 func (r *rs) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	conversion.ImportStateProjectIDClusterName(ctx, req, resp, "project_id", "name")
 }
 
-func mockedSDK(ctx context.Context, diags *diag.Diagnostics, timeout timeouts.Value) (*TFModel, bool) {
-	sdkResp, err := ReadClusterResponse()
+func (r *rs) createCluster(ctx context.Context, plan *TFModel, diags *diag.Diagnostics, resp *resource.CreateResponse) (summary, detail string) {
+	sdkReq := NewAtlasReq(ctx, plan, diags)
+	if diags.HasError() {
+		return "", ""
+	}
+	api := r.Client.AtlasV220240805.ClustersApi
```

Comment:
I think it's more clear to use api only for AtlasV2, i would use apiSomething for others similar to apiLegacy.  What's special in AtlasV220240805 that we can't use AtlasV2? so we can name it with that

Reply:
Will be resolved in https://jira.mongodb.org/browse/CLOUDP-286479